### PR TITLE
feat(Firma con IO): [SFEQS-1097] Router screen for signature request

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
In this PR has been added a router screen to manage different status for signature detail request.

https://user-images.githubusercontent.com/49342144/200808215-4a70198c-a02e-4af6-8ae5-ceeda6e86ceb.mp4

## List of changes proposed in this pull request
- Added backend api to get signature request detail view
- Added networking saga `handleGetSignatureRequestById`
- Added `FciRouterScreen`
- Added `ErrorComponent` and `SuccessComponent`
- Added a base `FciStackNavigator`
- Updated `AppStackNavigator`
- Added some tests for router screen and components

## How to test
- get `io-dev-server-api` and (if not yet merged) use https://github.com/pagopa/io-dev-api-server/pull/190
- enable `FCI_ENABLED` FF

### CODE REVIEW

File: ts/features/fci/saga/networking/handleGetSignatureRequestById.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// handle bonus list loading
+export function* handleGetSignatureRequestById(
+  getSignatureDetailViewById: ReturnType<
+    typeof BackendFciClient
+  >["getSignatureDetailViewById"],
+  action: ActionType<typeof fciSignatureRequestFromId["request"]>
+): SagaIterator {
+  try {
+    const getSignatureDetailViewByIdResponse: SagaCallReturnType<
```
Comment (francescopersico) [inline ts/features/fci/saga/networking/handleGetSignatureRequestById.ts:19]: Are you sure the return type is not inferred?

File: ts/features/fci/saga/networking/handleGetSignatureRequestById.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      typeof getSignatureDetailViewById
+    > = yield* call(getSignatureDetailViewById, { id: action.payload });
+    if (E.isLeft(getSignatureDetailViewByIdResponse)) {
```
Comment (francescopersico) [inline ts/features/fci/saga/networking/handleGetSignatureRequestById.ts:22]: Can we avoid using **if** (i know it is the old style already used in the app)?

File: ts/features/fci/saga/networking/handleGetSignatureRequestById.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      throw Error(
+        readablePrivacyReport(getSignatureDetailViewByIdResponse.left)
+      );
+    }
+    if (E.isRight(getSignatureDetailViewByIdResponse)) {
```
Comment (francescopersico) [inline ts/features/fci/saga/networking/handleGetSignatureRequestById.ts:27]: Not needed? (you already excluded left in the if above)

File: ts/features/fci/screens/FciRouterScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}>;
+
+const FciSignatureScreen = (
+  props: IOStackNavigationRouteProps<FciParamsList, "FCI_ROUTER">
+): React.ReactElement => {
+  const dispatch = useIODispatch();
+  const fciSignatureRequest = useIOSelector(fciSignatureRequestSelector);
+
+  React.useEffect(() => {
+    const signatureRequestId = props.route.params.signatureRequestId;
```
Comment (francescopersico) [inline ts/features/fci/screens/FciRouterScreen.tsx:25]: What you think about creating a const with signatureRequestId and than use it here and in the dependency array?

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/features/fci/saga/networking/handleGetSignatureRequestById.ts (hunk 2 above)
Comment (hevelius) [reply ts/features/fci/saga/networking/handleGetSignatureRequestById.ts:25]: Could you provide me a reference for the "new style" used in the app?

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/features/fci/components/SuccessComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import I18n from "../../../i18n";
+import imageExpired from "../../../../img/wallet/errors/payment-expired-icon.png";
+import hourglass from "../../../../img/pictograms/hourglass.png";
+import { SignatureRequestDetailView } from "../../../../definitions/fci/SignatureRequestDetailView";
+import FciDocumentsScreen from "../screens/valid/FciDocumentsScreen";
+import ErrorComponent from "./ErrorComponent";
+import GenericErrorComponent from "./GenericErrorComponent";
+
+const now = new Date();
```
Comment (francescopersico) [inline ts/features/fci/components/SuccessComponent.tsx:10]: Are you sure you want to create the Date here.
It will be never updated after app start.

**Review by francescopersico — CHANGES_REQUESTED** [review]

**Review by francescopersico — APPROVED** [review]

**Pull request merged**